### PR TITLE
Update tracking prompt accessibility

### LINKED ISSUE #2215: Opt-in usage tracking notice improvements
Looking a bit into the [recently added](https://github.com/WordPress/gutenberg/pull/2140) usage tracking notice, noticed a few things:

firefox

There are two "dismiss" buttons:

screen shot 2017-08-04 at 11 11 47

this is because of the combined effect of the component and the WordPress `makeNoticesDismissible()` (which runs on document ready). Since this notice appears on page load, `makeNoticesDismissible()` adds its own dismiss button.

Reducing the strings to translate is always a good thing. See `Dismiss this notice` vs. `Dismiss this notice.` (period at the end).

The order of the elements in the notice could be improved. Currently, the Yes and No button are placed before the text. Screen reader users will hear, in this order:
Yes
No
Can Gutenberg collect data ...
Dismiss this notice
(Dismiss this notice.)

Also from a logical perspective, I'd say available answers should go after the question.

All text is wrapped in `<div>`s. It would be nice to use slightly more semantic elements for text. The WordPress notices use paragraphs.

The italic text uses a `font-size: 90%;` which translates to a computed value of `11.7px`. This is definitely too small. I'd also propose to use values that translate to a computed value without rounding.

Minor: the rendering is different across browsers. I'd propose no action on this, just something to be aware of:

Chrome:

chrome

Firefox: 

firefox

this ie because bold text is rendered differently.

### PR DESCRIPTION
**Pull request by nylen**
Fixes #2215.

- [x] Remove the default notice dismissal behavior for this notice
- [x] Move the control buttons in the editor tracking prompt to after the content
- [x] Move the clarification message behind a "More info" button and a `Popover` component
- [x] Add focus styles and any needed `aria` attributes for the "More info" button

_edit: outdated screenshots removed; see below_

### CODE REVIEW

File: editor/enable-tracking-prompt/index.js [hunk 1]
```diff
@@ -18,48 +20,89 @@ import { removeNotice } from '../actions';
...
+						<span>{ __( 'More info' ) }</span>
+						{ this.state.showInfoPopover && (
+							<Popover
+								position="bottom right"
+								className="enable-tracking-prompt__clarification"
+							>
+								{ __( 'Usage data is completely anonymous, does not include your post content, and will only be used to improve the editor.' ) }
+							</Popover>
+						) }
+					</Button>
```
Comment (afercia) [inline editor/enable-tracking-prompt/index.js]: I'd move the popover out from the button and use `aria-expanded` on the button based on the showInfoPopover state. See for example post-visibility.

**Review by afercia — CHANGES_REQUESTED** [review]
Thanks @nylen ! just a coupe small things.

File: editor/enable-tracking-prompt/index.js (hunk 1 above)
Comment (nylen) [inline editor/enable-tracking-prompt/index.js]: I think then we need another container element with `position: relative` to hold the button and the popover.

Comment (afercia) [inline editor/enable-tracking-prompt/index.js]: Yep, I think that's the reason why [a wrapper element was added](https://github.com/WordPress/gutenberg/commit/9a7a34c4ad77514d61bc2c1ccb36adb5e53c16ed) around button and popover in the post-visibility.

**Comment by nylen** [comment]
Feedback addressed; focus styles added:

> 

Note that the `Popover` renders as a `div`, so it cannot be a child of a `p` element.  Otherwise, the following React warning occurs:

```
Warning: validateDOMNesting(...): <div> cannot appear as a descendant of <p>.
```

**Review by afercia — APPROVED** [review]

**Comment by afercia** [comment]
Thanks! :shipit:

**Pull request merged**